### ISSUE TITLE
Error if parameters are given in guard functions

### ISSUE THREAD

**Issue by bdemann** [demergent-labs/kybra#366]
When you are making a function guard, it seems like parameters for the function guard are not an option. It makes sense because, after all, who would provide the values for those parameters? On thought is that it could have the same parameters as the function it is guarding. I tested this in rust and it doesn't seem to work. 

This code gave the following error: 
```
fn guard_function(message: String) -> Result<(), String> {
    Err(format!("We are guarding: {}", message))
}

#[ic_cdk_macros::update(guard = "guard_function")]
fn update(message: String) -> () {
    CURRENT_MESSAGE_REF_CELL.with(|current_message_ref_cell| {
        *current_message_ref_cell.borrow_mut() = message;
    });
}
```

```
error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> canisters/rust/src/lib.rs:17:1
   |
17 | #[ic_cdk_macros::update(guard = "guard_function")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an argument of type `std::string::String` is missing
   |
note: function defined here
  --> canisters/rust/src/lib.rs:13:4
   |
13 | fn guard_function(message: String) -> Result<(), String> {
   |    ^^^^^^^^^^^^^^ ---------------
   = note: this error originates in the attribute macro `ic_cdk_macros::update` (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide the argument
   |
17 | #[ic_cdk_macros::update(guard = "guard_function")](/* std::string::String */)
   |                                                   +++++++++++++++++++++++++++

```

Which sounds a lot like "There isn't a way to get that parameter into the function call" if I am reading that correctly. Through honestly I'm not 100% sure about all of the details. 

At first when I was implementing the function guards in kybra I just passed a long the parameters without really thinking about it because they were easy to get thanks to the canister method functions that were already built. So I was able to verify that in kybra if the given parameters are passed a long then you will get those same error after it has been converted to rust.

I can't find any documentation explicitly stating that guard functions can't have parameters, but none of the examples show parameters, and we have this error. So I'm guessing not.

With all of that being said, we may want to consider providing a more formal error to a kybra user if they attempt to use a parameter with a function guard.

**Comment by bdemann** [comment]
As near as I can tell this has been fixed for quite a while, 6966da966cd42e897140143700dfde99a873eded was the earliest commit I could find with it fixed, but it looks like it was moving files around that caused me to not be able to go back further. I'll test it out to make sure but I think this is good to go.

**Comment by bdemann** [comment]
Confirmed, this is done
